logstash-core/lib/logstash/settings.rb

```diff
@@ -200,25 +225,61 @@ def coerce(value)
           raise ArgumentError.new("could not coerce #{value} into a boolean")
         end
       end
+    end
 
-      def set(value)
-        coerced_value = coerce(value)
-        validate(coerced_value)
-        @value = coerce(coerced_value)
-        @value_is_set = true
-        @value
+    class Numeric < Coercible
+      def initialize(name, default=nil, strict=true)
+        super(name, ::Numeric, default, strict)
+      end
+
+      def coerce(v)
+        return v if v.is_a?(::Numeric)
```

Comment:
Totally necessary the ```::Numeric```, however here I am on the fense of

* Using the same number and force people to use the `::` before the name
* Using an specific name that does not collide.

Can you share your pros and cons for going that way? this would be awesome to we can make sure to reduce the error rate when in a few weeks we need to change stuff here and use the `is_a?`, and alike, comparisons.